### PR TITLE
ISSUE-643 [APISIX] Add circuit breaker plugin

### PR DESCRIPTION
**Pull request by vttranlina**
resolve: https://github.com/linagora/tmail-backend/issues/643

### CODE REVIEW

**Review by quantranhong1999 — COMMENTED** [review]
Nothing to add

File: demo/apisix/conf/apisix.yaml [hunk 1]
```diff
@@ -86,9 +86,37 @@ services:
...
+            - 504
+          failures: 3  # should be var: JMAP_CIRCUIT_BREAKER_MAXERRORS
+        healthy:
+          successes: 1
+
   -
     id: jmap_no_auth_service
     upstream_id: jmap_upstream
+    plugins:
+      api-breaker:
```
Comment (Arsnael) [inline demo/apisix/conf/apisix.yaml:107]: The double declaration is redundant. I believe we could manage to declare the plugin only once. Look at my PR. Declaring a plugin and using `plugin_config_id`, even if other plugins are declared seem to work (which I like, avoid some useless verbosity like in krakend)

**Review by Arsnael — COMMENTED** [review]
No test? No video?

**Comment by vttranlina** [comment]
https://user-images.githubusercontent.com/81145350/234730400-148993e1-8748-42ab-aa32-3d73b8b1e300.mov

test video

**Review by Arsnael — APPROVED** [review]

**Review by quantranhong1999 — APPROVED** [review]

**Comment by Arsnael** [comment]
Rebase please

**Comment by psouquet** [comment]
Hi, I review this as cross team review and I had a small question: why sending a 503 only once ? Wouldn't it be better to always reply it until the request goes through ?  
Of course, feel free to disregard my comment, I'm just curious :)

**Comment by Arsnael** [comment]
> Hi, I review this as cross team review and I had a small question: why sending a 503 only once ? Wouldn't it be better to always reply it until the request goes through ? 

That's what happens already though :) Apisix will always return 503 to the client for as long as the circuit is opened. The test video might be a bit misleading though... 

Now I have looked at it again though, @vttranlina : 
-> my guess is you use the timeout for the purpose of your test (to generate errors quickly while pausing the container). I guess you have the circuit breaker plugin in place. But it would have been nice to show the conf of the plugin and not just only the timeout in the video :)

**Comment by Arsnael** [comment]
Rebased and squashed

**Pull request merged**